tools/lou_checkyaml.c

```diff
@@ -1004,21 +1009,24 @@ main(int argc, char *argv[]) {
 	}
 	yaml_event_delete(&event);
 
-	int has_next;
-	has_next = yaml_parser_parse(&parser, &event);
-
-	if (has_next && event.type == YAML_SCALAR_EVENT &&
-			!strcmp((const char *)event.data.scalar.value, "display")) {
-		display_table = read_table_value(&parser, event.start_mark.line + 1);
-		yaml_event_delete(&event);
-		has_next = yaml_parser_parse(&parser, &event);
-	}
-
-	if (!has_next) simple_error("table expected", &parser, &event);
+	if (!yaml_parser_parse(&parser, &event))
+		simple_error("table expected", &parser, &event);
 
 	int MAXTABLES = 150;
 	char *tables[MAXTABLES + 1];
-	while ((tables[0] = read_table(&event, &parser))) {
+	while (1) {
```

Comment:
Somehow I'm not such a fan of `while (1)` but maybe I'm splitting hairs

Reply:
Do you prefer the more criptic `for (;;)`?

Reply:
Haha, no but it seems that before it had a proper termination condition in the while loop and now it is hidden somewhere below in an `if (...) break`